Classes/Plugin/OaiPmh.php

```diff
@@ -958,14 +960,25 @@ protected function generateOutputForDocumentList(DocumentList $documentListSet)
             'AND `tx_dlf_documents`.`pid` = ? ' .
             'AND `tx_dlf_collections`.`pid` = ? ' .
             'AND `tx_dlf_relations`.`ident`="docs_colls" ' .
-            'AND ' . Helper::whereExpression('tx_dlf_collections') .
-            ' LIMIT ?';
+            'AND ' . Helper::whereExpression('tx_dlf_collections') . ' ' .
+            'GROUP BY `tx_dlf_documents`.`uid` ' .
+            'LIMIT ?';
+
+        $values = [
+            $documentsToProcess,
+            $this->conf['pages'],
+            $this->conf['pages'],
+            $this->conf['limit']
+        ];
+        $types = [
+            Connection::PARAM_INT_ARRAY,
+            Connection::PARAM_INT,
+            Connection::PARAM_INT,
+            Connection::PARAM_INT
+        ];
+        // Create a prepared statement for the passed SQL query, bind the given params with their binding types and execute the query
+        $documents = $connection->executeQuery($sql, $values, $types);
 
-        $documents = $connection->prepare($sql);
-        $documents->bindValue(1, implode(',', $documentsToProcess));
-        $documents->bindValue(2, (int)$this->conf['pages']);
-        $documents->bindValue(3, (int)$this->conf['pages']);
-        $documents->bindValue(4, (int)$this->conf['limit']);
         $documents->execute();
```

Comment:
```suggestion
```
`$connection->executeQuery()` already returns a database result object, so we don't need another `execute()` here.